source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -142,14 +185,20 @@ FilterConfigPerRoute::mergeProcessingMode(const FilterConfigPerRoute& less_speci
 
 FilterConfigPerRoute::FilterConfigPerRoute(const ExtProcPerRoute& config)
     : disabled_(config.disabled()), processing_mode_(initProcessingMode(config)),
-      grpc_service_(initGrpcService(config)) {}
+      grpc_service_(initGrpcService(config)),
+      untyped_forwarding_namespaces_(initUntypedForwardingNamespaces(config)),
+      typed_forwarding_namespaces_(initTypedForwardingNamespaces(config)),
+      untyped_receiving_namespaces_(initUntypedReceivingNamespaces(config)) {}
 
 FilterConfigPerRoute::FilterConfigPerRoute(const FilterConfigPerRoute& less_specific,
                                            const FilterConfigPerRoute& more_specific)
     : disabled_(more_specific.disabled()),
       processing_mode_(mergeProcessingMode(less_specific, more_specific)),
       grpc_service_(more_specific.grpcService().has_value() ? more_specific.grpcService()
-                                                            : less_specific.grpcService()) {}
+                                                            : less_specific.grpcService()),
+      untyped_forwarding_namespaces_(more_specific.untypedForwardingMetadataNamespaces()),
+      typed_forwarding_namespaces_(more_specific.typedForwardingMetadataNamespaces()),
+      untyped_receiving_namespaces_(more_specific.untypedReceivingMetadataNamespaces()) {}
```

Comment:
@jbohanon  I agree with getting rid of merge proto. That probably has unnecessary complexity that we don' need for now.   

However, I am not sure that we should let control plane decides it . At least it seems to be inconsistent with existing ext_proc behavior. e.g., `mergeProcessingMode`, `grpc_service_` above.  Can we just have similar simple logic : if most_specific has namespaces, use that; if most_specific doesn't , use less_specific.

This is probably not critical thing to this PR and I am ok with discussing it in a separate PR.  Small PR which can help review and move things faster is preferred and appreciated  .

Reply:
> However, I am not sure that we should let control plane do the merge . At least it seems to be inconsistent with existing ext_proc behavior. e.g., mergeProcessingMode, grpc_service_ above. Can we just have similar simple logic : if most_specific has namespaces, use that; if most_specific doesn't , use less_specific.

I might have worded this poorly. This is currently the behavior. What was meant by letting the control plane handle the merge is that the if a control plane implementation has abstractions over Envoy APIs, then during the translation down to Envoy API, it is the responsibility of the control plane to ensure that the most specific config defined is the desired state of the metadata namespaces config/overrides

Reply:
SGTM, Thanks

Reply:
After discussing with @tyxia I need to revisit the merge logic here to make sure nonexistent more-specific config doesn't blow out existing less-specific config

/wait